src/tests/testCacheManager.cc

```diff
@@ -208,11 +208,10 @@ testCacheManager::testParseUrl()
                             std::cerr << std::endl
                                       << "FAIL: " << mgrUrl
                                       << Debug::Extra << "error: should be rejected due to '" << param << "'" << std::endl;
-                            CPPUNIT_FAIL(err);
                         } catch (const TextException &e) {
-                            if (strcmp(e.what(), err) == 0)
-                                throw;
+                            continue; // success. caught bad input
                         }
+                        CPPUNIT_FAIL("bad result");
```

Comment:
Let's be more specific to help developers figure out what went wrong if this test case fails:

```suggestion
                        CPPUNIT_FAIL("failed to reject an invalid URL");
```